pkg/policy/trace/yaml.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/cilium/cilium/pkg/labels"
+	"io"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+)
+
+const (
+	// ReplicationController represents a Kubernetes ReplicationController kind.
+	ReplicationController = "ReplicationController"
+
+	// Deployment represents a Kubernetes Deployment kind.
+	Deployment            = "Deployment"
+
+	// ReplicaSet represents a Kubernetes ReplicaSet kind.
+	ReplicaSet            = "ReplicaSet"
+
+	// DefaultNamespace represents the default Kubernetes namespace.
+	DefaultNamespace      = "default"
+)
+
+// GetLabelsFromYaml iterates through the provided YAML file and for each
+// section  in the YAML, returns the labels or an error if the labels could not
+// be parsed.
+func GetLabelsFromYaml(file string) ([][]string, error) {
```

Comment:
Why not simply use kubernetes unmarshall the file?

Reply:
Please clarify what this means.

Reply:
I thought I had write more than that.
Why not simply use the kubernetes library to unmarshall the file? This https://github.com/kubernetes/apimachinery/blob/edfbcaadfa90c46ecb75bd3da453456085a1eec2/pkg/util/yaml/decoder.go#L207

Reply:
I am using that library, but I am using a `YAMLToJSONDecoder`.

Reply:
What I meant since the beginning is there is an official way of decoding kubernetes spec files. I thought it was the link I've provided but after digging a little more I think this is the one https://github.com/kubernetes/apimachinery/blob/edfbcaadfa90c46ecb75bd3da453456085a1eec2/pkg/runtime/serializer/codec_factory.go

With that package you'll be able to deserialize the file into an object ( see UniversalDeserializer that returns a Decoder https://github.com/kubernetes/apimachinery/blob/edfbcaadfa90c46ecb75bd3da453456085a1eec2/pkg/runtime/interfaces.go#L50 )

Reply:
What's the advantage of using the Kubernetes deserializer? Does it mean we have to take on additional dependencies? I would like to know what we gain before we continue to grow our binary size.